src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -311,11 +311,10 @@ private static CompletionItemRules MakeRule(bool importDirective, bool preselect
 
         protected override string GetInsertionText(CompletionItem item, char ch)
         {
-            if (ch == ';' && SymbolCompletionItem.GetShouldProvideParenthesisCompletion(item))
+            if (ch is ';' or '.' && SymbolCompletionItem.GetShouldProvideParenthesisCompletion(item))
```

Comment:
Just found This [test ](https://github.com/dotnet/roslyn/blob/7902b3be7b424a5072dee5cd4b153d7b1015584d/src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb#L1810)is blocking the dot as a special completion character

From its original [link ](https://github.com/dotnet/roslyn/issues/13527) and [PR](https://github.com/dotnet/roslyn/pull/13638), I feel it indicates that dot should not be a commit character under tuple syntax context. 

We could double and avoid adding parenthesis if it is under tuple context. But I don't want to make this so complicated. The original idea for using dot to add parenthesis is I feel it would be useful when typing a method call chain. Like `
"Hello".ToString().ToString().ToString()
`. But if the user found that it sometimes not works when typing tuple it would become puzzled.

I am ok to close this PR if we think this would introduce puzzling experience. 
Tag @genlu  for more idea here.

Reply:
I'm not following. The discussion you linked is about not commit when `:` is typed in tuple context, which makes sense since `:` is part of the "field name" syntax. But I can't think of any reason why allowing `.` to be a commit char would cause problem. Could you please explain?

Reply:
I am also not sure but that's what I got from the unit test.

Tag @jcouv for more info,
Hi Julien,
This PR is trying to introduce '.' (dot) as a completion commit character, and is blocked by this [test](https://github.com/dotnet/roslyn/blob/7902b3be7b424a5072dee5cd4b153d7b1015584d/src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb#L1810)
From the [link ](https://github.com/dotnet/roslyn/issues/13527)I saw you are the original owner of that test.

The context for this PR is if the user uses dot to commit a method, and the context type is not delegate, a parenthesis pair would be appended.

From the original issue link and PR title it seems like ':' (colon) should not be considered as a commit character under tuple context, but Gen and I can't get the idea why dot is not allowed. Do you think it would be OK to add dot as a commit character?

Reply:
I suspect that test was to capture a behavior that I noticed, but not necessarily to lock that behavior in (that test was added in PR was tuple completion which doesn't really care about `.`).